web_src/less/features/heatmap.less

```diff
@@ -1,9 +1,9 @@
 #user-heatmap {
   width: 100%;
+  display: inline;
   text-align: center;
   position: relative;
   min-height: 125px;
-  display: flex;
   align-items: center;
   justify-content: center;
```

Comment:
If you remove `display: flex` (which I don't agree with), you can also remove `align-items` and `justify-content` because those properties only apply to flexbox.

Please check if the placeholder seen in https://github.com/go-gitea/gitea/pull/13623#discussion_r526334214 is still centered. I still think the flexbox removal is unwarranted, it's the best method to center things.

Reply:
the error message could be flex-centered once the error occurs, maybe by defining another heatmap-error class or sth.
let me see if I can trigger an error to verify what it looks like now.

Reply:
You can add a `throw new Error("fail")` at a place like here:

https://github.com/go-gitea/gitea/blob/ad2a288622c7bbf53b8670e136e2e1418331b121/web_src/js/features/heatmap.js#L13

Reply:
> Please check if the placeholder seen in #13623 (comment) is still centered. I still think the flexbox removal is unwarranted, it's the best method to center things.

I still wouldn't say a centered error message is more important than a nicely aligned heatmap, don't know...
hopefully, we can do both.

Reply:
> the error message could be flex-centered

why not flex-center everything? I see no drawback.

Reply:
> > the error message could be flex-centered
> 
> why not flex-center everything including the svg? I see no drawback.

I do - it's in the screenshots and it's the reason I opened this issue.
It is partly solvable adding `{width,height}: 100%`.
But that still doesn't scale so well on smaller displays.
The inline variant aligns the legend and stats on top of all of the above.

Reply:
the error message is nicely justified
![image](https://user-images.githubusercontent.com/61180606/99667061-1dd0f480-2a6c-11eb-8972-234f9d514f60.png)

Reply:
which is not perfect but not the worst either and considering it's an edge case and the fact that when shown it *is* shown nicely, I'd think it's manageable.

Reply:
@silverwind  I think I got a remedy for your concern.

Reply:
what do you say on https://github.com/go-gitea/gitea/pull/13637/commits/474119d7687f217c8335c55f7071fc5d7dfe1d61 ?
is that alright?

Reply:
Not sure yet, need to test later.

Reply:
`""`-->`''` in https://github.com/go-gitea/gitea/commit/650964be480264897848e223a65ff499083bdbc0
posted a screenshot below, too.